### PR TITLE
Additional properties to Person, ref #961

### PR DESCRIPTION
**Pull request by awead**
Adds email, PSU id, and Orcid id to the Person model so we can store
these values when users want to add them.

### CODE REVIEW

File: app/controllers/persons_controller.rb [hunk 1]
```diff
@@ -4,8 +4,26 @@ class PersonsController < ApplicationController
...
+
+    def search_results(query)
+      aliases = ActiveFedora::SolrService.query(
+        "display_name_tesim:#{query}*",
+        fq: 'has_model_ssim:Alias',
+        fl: ['id', 'display_name_tesim', 'name_ssim'],
+        qt: 'select',
+        rows: 10
+      )
+      aliases.each do |aliaz|
```
Comment (carolyncole) [inline app/controllers/persons_controller.rb]: @awead how about agent_alias?  We are never going to know to search for aliaz if we are looking at the code.

File: app/controllers/persons_controller.rb [hunk 2]
```diff
@@ -4,8 +4,26 @@ class PersonsController < ApplicationController
...
+        person = ActiveFedora::SolrService.query(
+          "id:#{aliaz['name_ssim'].first}",
```
Comment (carolyncole) [inline app/controllers/persons_controller.rb]: @awead how is the name the ID?  That seems strange

File: app/forms/creator_form.rb [hunk 3]
```diff
@@ -30,6 +30,21 @@ def given_name
     end
   end
 
+  def psu_id
+    return unless read_only_name?
```
Comment (carolyncole) [inline app/forms/creator_form.rb]: I'm wondering based on the fact that we are using read_only_name? for things that are not related to name if we should change that to just read_only?

File: spec/factories/persons.rb [hunk 4]
```diff
@@ -5,4 +5,12 @@
     sequence(:given_name) { |n| "First Name #{n}" }
     sequence(:sur_name) { |n| "Last Name #{n}" }
   end
+
+  trait :with_complete_metadata do
+    given_name 'Johnny C.'
+    sur_name   'Lately'
+    email      'newkid@example.com'
+    psu_id     'jcl81'
+    orcid_id   '00123445'
```
Comment (carolyncole) [inline spec/factories/persons.rb:14]: I wonder if we care about validating the orcid.

**Review by carolyncole — CHANGES_REQUESTED** [review]

File: app/controllers/persons_controller.rb (hunk 2 above)
Comment (awead) [reply app/controllers/persons_controller.rb]: @cam156 This one got me too. It's generated by ActiveFedora's solr name stuff. Specifically, when you define the relationship using the `name` predicate here:

https://github.com/psu-stewardship/scholarsphere/blob/user-object-961/app/models/alias.rb#L6

I would have expected `person_ssim`, but I guess not.

File: spec/factories/persons.rb (hunk 4 above)
Comment (awead) [reply spec/factories/persons.rb:14]: @cam156 dunno. I imagine we should. We put that in another ticket and validate by some concrete means, like length, etc., and then look at a future option that could query the orchid service directly, like an autocomplete or just a call back that's done at some point.

File: .rubocop_todo.yml [hunk 5]
```diff
@@ -1,6 +1,12 @@
-require: rubocop-rspec
```
Comment (carolyncole) [inline .rubocop_todo.yml:1]: @awead did you intend to remove rubocop-rspec?

Comment (awead) [reply .rubocop_todo.yml:1]: hmm. no. I think auto-gen did that.

Comment (awead) [reply .rubocop_todo.yml:1]: @cam156 `--auto-gen-config` seems to have done that. I'm not sure why it's showing up now, but the require is in `.rubocop_yml` so it doesn't need to be in the todo file.

**Comment by awead** [comment]
@cam156 I took a look at the coveralls changes, and the missing coverage doesn't seem to be related to any of the changes in this PR.

**Comment by awead** [comment]
@cam156 oohhh-kay.... now it says coverage increased. I rebased it, so maybe that was it.

**Pull request merged**